src/puppetlabs/puppetdb/mq_listener.clj

```diff
@@ -109,6 +122,17 @@
       (swap! metrics assoc-in storage-path
              (create-metrics [(keyword command) (keyword (str version))])))))
 
+(defn initialize-depth!
+  [q]
+  (let [queue-dir (str (.toString (:directory q)) "/q")]
```

Comment:
I'm not convinced that this is necessary. We already get this information from stockpile on startup, why not hook in there so we don't have to reread all of the files on disk?